ludwig/data/dataset/ray.py

```diff
@@ -72,15 +72,33 @@ def __init__(
         #     return df
         # self.ds = self.ds.map_batches(to_tensors, batch_format="pandas")
 
-    def pipeline(self, shuffle=True, fully_executed=True) -> DatasetPipeline:
+    def pipeline(
+        self,
+        shuffle: bool = True,
+        fully_executed: bool = True,
+        enable_windowing: bool = False,
+        window_size_bytes: Optional[int] = None,
+    ) -> DatasetPipeline:
+        """
+        Args:
+            shuffle: If true, the entire dataset is shuffled in memory before batching.
+            fully_executed: If true, force full evaluation of the Ray Dataset by loading all blocks into memory.
+            enable_windowing: If true, split dataset into windows of fixed memory size.
+            window_size_bytes: If not None, specifies the window size in bytes for the dataset.
+        """
         if not fully_executed and not _ray112:
             raise ValueError(f"Cannot set fully_execute=False in ray {ray.__version__}")
 
         if fully_executed and _ray112:
             # set instance state so calls to __len__ will also use the fully_executed version
             self.ds = self.ds.fully_executed()
 
-        pipe = self.ds.repeat()
+        if not enable_windowing:
+            pipe = self.ds.repeat()
+        else:
+            if window_size_bytes is None:
```

Comment:
An alternative to requiring `window_size_bytes` to always be set is to calculate window size based on the following variables:
- Total object store memory in the ray cluster, and
- If running from within a tune trial and piping through the max concurrent trials parameter.
Not sure if that would be overkill tho so I've implemented a simpler solution for now.